Feature/auction amount summary

Velenir opened the pull request:
Changes:

1. Added `AuctionAmountSummary` component that looks like a static `TokenPair` with already set selling and buying amounts.
2. `TokenPair` state now includes `sellAmount` field, which together with closing price makes up estimated `buyAmount`
2. Made `AuctionSellingGetting` into a controlled component, so that we can pull `sellAmount` whenever we need it, e.g. in `AuctionAmountSummary`, and later when submitting sell order.
3. Added `TokenOverlay` to `PANEL 2`
4. misc fixes

Velenir committed:
added default value for "to" prop on ButtonCTA

Velenir committed:
created AuctionAmountSummary component

Velenir committed:
connected AuctionAmountSummary to store

Velenir committed:
fixed setting wrong prop in tokenOverlay reducer

Velenir committed:
added sellAmount field to TokenPair state

Velenir committed:
added setSellTokenAmount action

Velenir committed:
added reducer for setSellTokenAmount action

Velenir committed:
changes to accomodate new sellAmount field in TokenPair state

Velenir committed:
changed AuctionAmountSummary container mapping to pass sellAmount to the component

Velenir committed:
changed AuctionSellingGetting into a controlled component to keep sellAmount in store

Velenir committed:
changes to accomodate AuctionSellingGetting as a controlled component

Velenir committed:
replaced TokenPair component with AuctionAmountSummary on PANEL 3

Velenir committed:
added notes for future

Velenir committed:
added TokenOverlay to OrderPanel

Velenir committed:
added todo notes

Velenir committed:
added Link component in place of anchor to allow navigation without page reload

Velenir committed:
fixed storybook breaking when Link component is outside of Router

Velenir committed:
brought back changing ratio knob to control buyAmount

Velenir committed:
fixed children components not having key props in array

Velenir committed:
replaced TokenPair component with AuctionAmountSummary on PANEL 3

Velenir committed:
added TokenOverlay component to PANEL 2

Velenir committed:
fixed wrong knob name

W3stside commented on src/components/OrderPanel/index.tsx:
```diff
@@ -16,6 +17,7 @@ interface OrderPanelProps {
   <AuctionContainer auctionDataScreen="amount">
+    <TokenOverlay />
     <AuctionHeader backTo="/">
       Token Auction ${sellToken}/${buyToken}
```
templating issue, you forgot the templating quotes

Velenir committed:
fixed template literal typo

Velenir committed:
added story for TokenPair in PANEL 2

Velenir committed:
fixed AuctionAmountSummary component name

Velenir committed:
added a story for AuctionAmountSummary component

Velenir committed:
fixed visual centering of arrow element

Velenir committed:
changed TokenPair to accept primitives of tokens and tokenBalances instead of objects

that way there'll potentially be fewer rerenders caused by  immutable tokenPair or tokenBalance states' modifications,
also it now has the same props as AuctionAmountSummary
